diff --git a/src/runtime/server/fetch.js b/src/runtime/server/fetch.js
index ff4a1fa527e2203883d6001a6992eab76203be68..56328081fc40a862618860a03b4c2d01cae0bdbe 100644
--- a/src/runtime/server/fetch.js
+++ b/src/runtime/server/fetch.js
@@ -64,7 +64,7 @@ export function create_fetch({ event, options, manifest, state, get_cookie_heade
 					// leading dot prevents mydomain.com matching domain.com
 					// Do not forward other cookies for "credentials: include" because we don't know
 					// which cookie belongs to which domain (browser does not pass this info)
-					if (`.${url.hostname}`.endsWith(`.${event.url.hostname}`) && credentials !== 'omit') {
+					if ((event.url.hostname.includes('santiment.net') || `.${url.hostname}`.endsWith(`.${event.url.hostname}`)) && credentials !== 'omit') {
 						const cookie = get_cookie_header(url, request.headers.get('cookie'));
 						if (cookie) request.headers.set('cookie', cookie);
 					}
diff --git a/src/runtime/server/page/load_data.js b/src/runtime/server/page/load_data.js
index 75cf68d335e8f5013b9c8195508c43420a9f4950..3488ed8aaf76f580c679ed5132abe4ea1f9f27ad 100644
--- a/src/runtime/server/page/load_data.js
+++ b/src/runtime/server/page/load_data.js
@@ -232,7 +232,9 @@ export function create_universal_fetch(event, state, fetched, csr, resolve_opts)
 		let response = await event.fetch(input, init);
 
 		const url = new URL(input instanceof Request ? input.url : input, event.url);
-		const same_origin = url.origin === event.url.origin;
+		const same_domain = url.hostname.includes('santiment.net');
+const same_origin = same_domain || url.origin === event.url.origin;
+        
 
 		/** @type {import('types').PrerenderDependency} */
 		let dependency;
