diff --git a/src/exports/vite/index.js b/src/exports/vite/index.js
index 94f242a1e468646c962375aee90e94a2f07b06c2..456ef17d2f4886a7463c134855424c9b194e9f22 100644
--- a/src/exports/vite/index.js
+++ b/src/exports/vite/index.js
@@ -490,6 +490,8 @@ function kit({ svelte_config }) {
 				const ssr = /** @type {boolean} */ (config.build?.ssr);
 				const prefix = `${kit.appDir}/immutable`;
 
+      	Object.assign(config.define, ssr ? config.serverDefines : config.clientDefines)
+
 				/** @type {Record<string, string>} */
 				const input = {};
 
diff --git a/src/runtime/server/page/load_data.js b/src/runtime/server/page/load_data.js
index bcc2443aede63b92c9ff99b367d44b86b538b5d6..17c36c66aa66ec88a193e628fa5cc0efb97c8bda 100644
--- a/src/runtime/server/page/load_data.js
+++ b/src/runtime/server/page/load_data.js
@@ -171,6 +171,8 @@ export async function load_data({
 	}
 
 	const result = await node.universal.load.call(null, {
+		request: event.request,
+		getClientAddress: event.getClientAddress, 
 		url: event.url,
 		params: event.params,
 		data: server_data_node?.data ?? null,