diff --git a/src/exports/vite/index.js b/src/exports/vite/index.js
index 06b1977c2b82c1dc3c68048cde90b4ad33e16dbd..b452b08224cd8ee9b3cbdc0c0dfe508f57ebbc89 100644
--- a/src/exports/vite/index.js
+++ b/src/exports/vite/index.js
@@ -550,6 +550,7 @@ async function kit({ svelte_config }) {
 			if (is_build) {
 				const ssr = /** @type {boolean} */ (config.build?.ssr);
 				const prefix = `${kit.appDir}/immutable`;
+        Object.assign(config.define, ssr ? config.serverDefines : config.clientDefines)
 
 				/** @type {Record<string, string>} */
 				const input = {};
diff --git a/src/runtime/server/fetch.js b/src/runtime/server/fetch.js
index ff4a1fa527e2203883d6001a6992eab76203be68..9c89bfbda92343266165bfacc749d7df8eb72902 100644
--- a/src/runtime/server/fetch.js
+++ b/src/runtime/server/fetch.js
@@ -64,7 +64,7 @@ export function create_fetch({ event, options, manifest, state, get_cookie_heade
 					// leading dot prevents mydomain.com matching domain.com
 					// Do not forward other cookies for "credentials: include" because we don't know
 					// which cookie belongs to which domain (browser does not pass this info)
-					if (`.${url.hostname}`.endsWith(`.${event.url.hostname}`) && credentials !== 'omit') {
+					if ((url.hostname.includes(process.env.CORS_HOSTNAME) || `.${url.hostname}`.endsWith(`.${event.url.hostname}`)) && credentials !== 'omit') {
 						const cookie = get_cookie_header(url, request.headers.get('cookie'));
 						if (cookie) request.headers.set('cookie', cookie);
 					}
diff --git a/src/runtime/server/page/load_data.js b/src/runtime/server/page/load_data.js
index 75cf68d335e8f5013b9c8195508c43420a9f4950..fe5e73a932e32d347b8ef0031b80bc0630aa8bd3 100644
--- a/src/runtime/server/page/load_data.js
+++ b/src/runtime/server/page/load_data.js
@@ -190,6 +190,9 @@ export async function load_data({
 	}
 
 	const result = await node.universal.load.call(null, {
+       		request: event.request,
+          getClientAddress: event.getClientAddress,
+          
 		url: event.url,
 		params: event.params,
 		data: server_data_node?.data ?? null,
@@ -232,7 +235,9 @@ export function create_universal_fetch(event, state, fetched, csr, resolve_opts)
 		let response = await event.fetch(input, init);
 
 		const url = new URL(input instanceof Request ? input.url : input, event.url);
-		const same_origin = url.origin === event.url.origin;
+		const same_domain = url.hostname.includes(process.env.CORS_HOSTNAME);
+const same_origin = same_domain || url.origin === event.url.origin;
+        
 
 		/** @type {import('types').PrerenderDependency} */
 		let dependency;
