diff --git a/files/handler.js b/files/handler.js
index 91cbd38b988019d261996f1fe0be600a941481a5..59b51c5f410b08e1835e2d4dbc898c2578c4094e 100644
--- a/files/handler.js
+++ b/files/handler.js
@@ -1205,6 +1205,7 @@ function serve(path, client = false) {
 					// only apply to build directory, not e.g. version.json
 					if (pathname.startsWith(`/${manifest.appPath}/immutable/`) && res.statusCode === 200) {
 						res.setHeader('cache-control', 'public,max-age=31536000,immutable');
+        if(pathname.endsWith('.woff2')) res.setHeader('Access-Control-Allow-Origin', '*')
 					}
 				})
 		})
