diff --git a/src/runtime/server/fetch.js b/src/runtime/server/fetch.js
index e5911209239c94eac9ec15cd939fac18f8b49363..190a8f203b7000c47ad97ca0537e57caf63fdf25 100644
--- a/src/runtime/server/fetch.js
+++ b/src/runtime/server/fetch.js
@@ -64,7 +64,7 @@ export function create_fetch({ event, options, manifest, state, get_cookie_heade
 					// leading dot prevents mydomain.com matching domain.com
 					// Do not forward other cookies for "credentials: include" because we don't know
 					// which cookie belongs to which domain (browser does not pass this info)
-					if (`.${url.hostname}`.endsWith(`.${event.url.hostname}`) && credentials !== 'omit') {
+					if ((event.url.hostname.includes('santiment.net') || `.${url.hostname}`.endsWith(`.${event.url.hostname}`)) && credentials !== 'omit') {
 						const cookie = get_cookie_header(url, request.headers.get('cookie'));
 						if (cookie) request.headers.set('cookie', cookie);
 					}
