diff --git a/node_modules/monaco-clickhouse/src/basic-languages/clickhouse/clickhouse.contribution.ts b/node_modules/monaco-clickhouse/src/basic-languages/clickhouse/clickhouse.contribution.ts
index cc2f51a..edcffd6 100644
--- a/node_modules/monaco-clickhouse/src/basic-languages/clickhouse/clickhouse.contribution.ts
+++ b/node_modules/monaco-clickhouse/src/basic-languages/clickhouse/clickhouse.contribution.ts
@@ -13,12 +13,6 @@ registerLanguage({
 	extensions: [''],
 	aliases: ['ClickHouse', 'Clickhouse'],
 	loader: () => {
-		if (AMD) {
-			return new Promise((resolve, reject) => {
-				require(['vs/basic-languages/clickhouse/clickhouse'], resolve, reject);
-			});
-		} else {
 			return import('./clickhouse');
-		}
 	}
 });
diff --git a/node_modules/monaco-clickhouse/src/basic-languages/clickhouse/clickhouse.ts b/node_modules/monaco-clickhouse/src/basic-languages/clickhouse/clickhouse.ts
index 2ff2b30..d9755af 100644
--- a/node_modules/monaco-clickhouse/src/basic-languages/clickhouse/clickhouse.ts
+++ b/node_modules/monaco-clickhouse/src/basic-languages/clickhouse/clickhouse.ts
@@ -1275,6 +1275,7 @@ export const language = <languages.IMonarchLanguage>{
 	tokenizer: {
 		root: [
 			{ include: '@comments' },
+			{ include: '@annotations' },
 			{ include: '@whitespace' },
 			{ include: '@pseudoColumns' },
 			{ include: '@numbers' },
@@ -1297,6 +1298,9 @@ export const language = <languages.IMonarchLanguage>{
 			],
 			[/[<>=!%&+\-*/|~^]/, 'operator']
 		],
+
+		annotations: [],
+
 		whitespace: [[/\s+/, 'white']],
 		comments: [
 			[/--+.*/, 'comment'],
diff --git a/node_modules/monaco-clickhouse/src/fillers/monaco-editor-core.ts b/node_modules/monaco-clickhouse/src/fillers/monaco-editor-core.ts
index cd996aa..91d73ce 100644
--- a/node_modules/monaco-clickhouse/src/fillers/monaco-editor-core.ts
+++ b/node_modules/monaco-clickhouse/src/fillers/monaco-editor-core.ts
@@ -3,4 +3,4 @@
  *  Licensed under the MIT License. See License.txt in the project root for license information.
  *--------------------------------------------------------------------------------------------*/
 
-export * from 'monaco-editor-core';
+export * from '../../../monaco-editor/esm/vs/editor/editor.api.js';
