diff --git a/lib/turndown.es.js b/lib/turndown.es.js
index 5e0f672a08d4d068cdd444f92b00026e2f21857c..cb16a8302d9f02f50223ca9c144a57ed7bd62347 100644
--- a/lib/turndown.es.js
+++ b/lib/turndown.es.js
@@ -1,3 +1,5 @@
+import domino from '@mixmark-io/domino'
+
 function extend (destination) {
   for (var i = 1; i < arguments.length; i++) {
     var source = arguments[i];
@@ -580,7 +582,6 @@ function createHTMLParser () {
   var Parser = function () {};
 
   {
-    var domino = require('@mixmark-io/domino');
     Parser.prototype.parseFromString = function (string) {
       return domino.createDocument(string)
     };
